# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ShareableCanvasRenderer.cpp
# Commit: e3051b96067a
# Full Hash: e3051b96067a480b020bd8c984a54491665c765e
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2021-11-11 15:25:44
# Regressor Bug: 1739448
# File Overlap Count: 1
# Description:
#   Bug 1739448 - Implement a prototype WebGL-backed Canvas2D. r=gfx-reviewers,nical,jrmuizel
#   
#   This mainly provides DrawTargetWebgl, which implements the subset of the DrawTarget
#   API necessary for integration with CanvasRenderingContext2D. It translates them to
#   suitable commands for its internal ClientWebGLContext, which then manages remoting
# ==============================================================================

diff -r 66c30950518d -r e3051b96067a gfx/layers/ShareableCanvasRenderer.cpp
--- a/gfx/layers/ShareableCanvasRenderer.cpp	Thu Nov 11 07:16:57 2021 +0000
+++ b/gfx/layers/ShareableCanvasRenderer.cpp	Thu Nov 11 07:16:58 2021 +0000
@@ -134,6 +134,11 @@
   // -
 
   const auto fnGetExistingTc = [&]() -> RefPtr<TextureClient> {
+    if (webgl) {
+      const auto desc = webgl->GetFrontBuffer(nullptr);
+      if (!desc) return nullptr;
+      return GetFrontBufferFromDesc(*desc, flags);
+    }
     if (provider) {
       if (!provider->SetKnowsCompositor(forwarder)) {
         gfxCriticalNote << "BufferProvider::SetForwarder failed";
@@ -142,12 +147,7 @@
 
       return provider->GetTextureClient();
     }
-
-    if (!webgl) return nullptr;
-
-    const auto desc = webgl->GetFrontBuffer(nullptr);
-    if (!desc) return nullptr;
-    return GetFrontBufferFromDesc(*desc, flags);
+    return nullptr;
   };
 
   // -